src/lib/comms/sol-lwm2m.c

```diff
@@ -3976,6 +4097,26 @@ sol_lwm2m_client_add_object_instance(struct sol_lwm2m_client *client,
     r = setup_instance_resource(client, ctx, instance, false);
     SOL_INT_CHECK_GOTO(r, < 0, err_exit);
 
+    if (client->supports_access_control &&
+        ctx->obj->id != SECURITY_SERVER_OBJECT_ID &&
+        ctx->obj->id != ACCESS_CONTROL_OBJECT_ID) {
+        //Since this API is expected to be used only in Factory Bootstrap mode,
+        // the owner of this Access Control Object Instance will be the
+        // Bootstrap Server, as well as the only server allowed to perform
+        // any (and all) operations, so there's no need for an ACL Resource
+        r = setup_access_control_object_instance_for_instance(client,
+            ctx->obj->id, instance->id, UINT16_MAX, NULL, false);
+
+        if (r == 0) {
+            SOL_DBG("Security Object Instance and Access Control Object Instance "
+                " does not need an Access Control Object Instance nor ACLs");
+        } else if (r != SOL_COAP_RESPONSE_CODE_CHANGED && r != SOL_COAP_RESPONSE_CODE_CREATED) {
+            SOL_WRN("Failed to create Access Control Object Instance for Object /%"
+                PRIu16 "/%" PRIu16, ctx->obj->id, instance->id);
+            return -ENOENT;
```

Comment:
-ENOENT, we should change the return code in here. ANd we must also remove the instance that was added.